lib/util/transformer/length-based-splitter.js

```diff
@@ -48,8 +58,37 @@ export class LengthBasedSplitter extends Stream.Transform {
     }
 
     const messageLength = this._readLength(this._frameBuffer, this.lengthFieldOffset, this.littleEndian);
+    
+    // Debug logging for frame sizes
+    if (DEBUG_CONFIG.enableFrameSizeLogging) {
+      log.debug(`Frame size detected: ${messageLength} bytes, max allowed: ${this.maxFrameLength} bytes`);
+    }
+    
+    // Log large frames for investigation
+    if (DEBUG_CONFIG.enableLargeFrameLogging && messageLength > DEBUG_CONFIG.largeFrameThreshold) {
+      log.warn(`Large frame detected: ${messageLength} bytes (${(messageLength / (1024 * 1024)).toFixed(2)} MB)`);
+      log.warn(`Service: ${this.serviceName}, Max allowed: ${this.maxFrameLength} bytes`);
+      
+      // Log memory usage if monitoring is enabled
+      if (DEBUG_CONFIG.enableMemoryMonitoring) {
+        const memUsage = process.memoryUsage();
+        log.warn(`Memory usage - RSS: ${(memUsage.rss / (1024 * 1024)).toFixed(2)} MB, Heap: ${(memUsage.heapUsed / (1024 * 1024)).toFixed(2)} MB`);
```

Comment:
consider using the [toReadableSizeString](https://github.com/appium/appium/blob/df56c667e8b1d144ba0310a9567581641c0d4347/packages/support/lib/util.js#L238) helper